pipelines/lstm.py

```diff
@@ -0,0 +1,131 @@
+import os
+from tempfile import NamedTemporaryFile
+
+import polars as pl
+import tensorflow
+from keras import layers, losses, metrics, models, optimizers
+from keras.utils import timeseries_dataset_from_array
+from prefect import task
+from wandb.keras import WandbMetricsLogger
+
+import wandb
+from pipelines.types import TimeWindow
+
+
+@task
+def shape_timeseries_dataset(
+    *,
+    data: pl.DataFrame,
+    features: list[str],
+    close_price_index: int,
+    window: TimeWindow,
+    stride: int = 1,
+    shuffle: bool = False,
+    batch_size: int = 8,
+) -> tensorflow.data.Dataset:
+    def split_window(
+        data: tensorflow.Tensor,
+    ) -> tensorflow.data.Dataset:
+        input_slice = slice(0, window.input)
+        labels_slice = slice(window.input, None)
+
+        inputs = data[:, input_slice, :]
+        labels = data[:, labels_slice, :]
+
+        labels = tensorflow.stack(
+            [labels[:, :, close_price_index]],
+            axis=-1,
+        )
+
+        inputs.set_shape([None, window.input, None])
+        labels.set_shape([None, window.output, None])
+
+        return (inputs, labels)
+
+    return timeseries_dataset_from_array(
+        data=data.select(features).to_numpy(),
+        targets=None,
+        sequence_length=window.length,
+        sequence_stride=stride,
+        shuffle=shuffle,
+        batch_size=batch_size,
+    ).map(split_window)
+
+
+@task
+def create(label_count, window: TimeWindow):
```

Comment:
Could we name this `create_model` or do you have a preference for it this way?

Reply:
`create_model` is fine with me, I was just using it like

```
from pipelines import lstm
```

so that each call would be like `<<model type>>.<<function>>`

like

```
lstm.create
lstm.train
```

etc, but `create_model` is probably even more clear, will change